src/test/java/org/cactoos/iterator/IteratorOfLongsTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
+
+import java.util.NoSuchElementException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+
+/**
+ * Tests for {@link IteratorOfLongs}.
+ *
+ * @since 0.32
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class IteratorOfLongsTest {
+    @Test
+    public void emptyIteratorDoesNotHaveNext() {
+        MatcherAssert.assertThat(
+            "hasNext is true for empty iterator.",
+            new IteratorOfLongs().hasNext(),
+            new IsEqual<>(false)
+        );
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void emptyIteratorThrowsException() {
+        new IteratorOfLongs().next();
+    }
+
+    @Test
+    public void nonEmptyIteratorDoesNotHaveNext() {
+        MatcherAssert.assertThat(
+            "hasNext is true for fully traversed iterator.",
+            this.iteratorWithFetchedElements().hasNext(),
+            new IsEqual<>(false)
+        );
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void nonEmptyIteratorThrowsException() {
+        this.iteratorWithFetchedElements().next();
+    }
+
+    private IteratorOfLongs iteratorWithFetchedElements() {
```

Comment:
@victornoel this should be static method, but better approach would be to remove it completely and move its implementation back to tests using different iterator size in each test. This would remove dependency between tests and provide more test diversity.

Reply:
I agree, I also did the same for the other `IteratorOfXXX` tests (actually some of them were already as you propose).